x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/Join.java

```diff
@@ -126,7 +117,7 @@ public static List<Attribute> computeOutput(List<Attribute> leftOutput, List<Att
             case LEFT -> {
                 // Right side becomes nullable.
                 List<Attribute> fieldsAddedFromRight = removeCollisionsWithMatchFields(rightOutput, matchFieldSet, matchFieldNames);
-                yield mergeOutputAttributes(makeNullable(makeReference(fieldsAddedFromRight)), leftOutput);
```

Comment:
I believe this nifty workaround was used to prevent extraction from the source - now it's no longer needed (and furthermore creates problems in the planner).